### PR TITLE
Deduplicate `is_server_notices_room`.

### PR DESCRIPTION
**Pull request by reivilibre**
Noticed this was duplicated and thought we may as well cut it out.

### CODE REVIEW

File: synapse/handlers/message.py [hunk 1]
```diff
@@ -752,20 +752,14 @@ async def _is_exempt_from_privacy_policy(
         if builder.type == EventTypes.Member:
             membership = builder.content.get("membership", None)
             if membership == Membership.JOIN:
-                return await self._is_server_notices_room(builder.room_id)
+                return await self.hs.get_room_member_handler().is_server_notice_room(
```
Comment (DMRobertson) [inline synapse/handlers/message.py:755]: This introduces a dependency `MessageHandler -> RoomMemberHandler`.

Does it make sense for handlers to interdepend?

(I expect this is no worse than we do elsewhere, so this is probably fine)

**Review by DMRobertson — APPROVED** [review]
SGTM but would still be interested on your thoughts re: that comment

File: synapse/handlers/message.py (hunk 1 above)
Comment (reivilibre) [reply synapse/handlers/message.py:755]: This is one of those things that I don't know the answer to myself, actually. It seems objectively worse to have the redundancy though, but maybe I'll ask around...

Comment (erikjohnston) [reply synapse/handlers/message.py:755]: We try and avoid it where possible, as we need to make sure that `RoomMemberHandler` is initialised before we start the homeserver (as handlers may raise on config errors etc), and doing inline `hs.get_*` obviously means we can't necessarily guarantee that if all *other* usages also get inlined. 

In this particular case: it feels like `is_server_notices_room` could either be moved into `ServerNoticesManager` or into the data store?

Comment (reivilibre) [reply synapse/handlers/message.py:755]: I never know what the distinction is meant to be between managers and handlers, but a note:

`ServerNoticesManager` calls into `RoomMemberHandler` (it takes a reference to it in `__init__` too).

If I move the function into there, we'll still have a circular dependency.

I didn't really realise that `get_*` was doing lazy initialisation.

I'll try putting it into the data store.

File: synapse/storage/databases/main/roommember.py [hunk 2]
```diff
@@ -504,6 +506,14 @@ async def check_local_user_in_room(self, user_id: str, room_id: str) -> bool:
 
         return membership == Membership.JOIN
 
+    async def is_server_notice_room(self, room_id: str) -> bool:
+        if self._server_notices_mxid is None:
```
Comment (erikjohnston) [inline synapse/storage/databases/main/roommember.py:510]: Cough docstring cough

**Review by erikjohnston — APPROVED** [review]

File: synapse/storage/databases/main/roommember.py (hunk 2 above)
Comment (reivilibre) [reply synapse/storage/databases/main/roommember.py:517]: ...if it makes you happy :)

**Pull request merged**